app/main.py

```diff
@@ -1,3 +1,61 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float | Distance) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
```

Comment:
Issue: The `__add__` method does not handle unsupported types. According to the checklist, you should raise a `TypeError` if `other` is not a supported type.

Reply:
Можна прибрати перевірку `elif isinstance(other, (int, float)):` і для решти випадків просто повертати `return Distance(self.km + other)`
```suggestion
    def __add__(self, other: int | float | Distance) -> "Distance":
        if isinstance(other, Distance):
            return Distance(self.km + other.km)
        return Distance(self.km + other)
```